### ISSUE TITLE
Hardcoded build paths cause errors when building for another platform

### ISSUE THREAD

**Issue by markubiak** [trezor/cython-hidapi#54]
Hello,

While using cython-hidapi on an embedded device, I came across a build issue. The setup.py contains hardcoded additions to the include path such as ``/usr/include/libusb-1.0`` and ``/usr/include/hidapi``. This, as it should, caused build errors since host x86 libraries were being used when the target device is armv7.

My temporary fix for this was to remove all extra include paths from setup.py, update chid.pyx to include ``"hidapi/hidapi.h"`` instead of just ``"hidapi.h"``, and build with --with-system-hidapi. Works great. However, it breaks normal build (also building hidapi) as that whole submodule requires those include paths to build.

I'm not certain why building hidapi is necessary instead of just requiring it to be installed (it's on every major and minor distro and build system), is it something to do with cython? If possible, I suggest getting rid of the submodule altogether and updating required dependencies. Either way, I've made a fork and am looking to upstream this.

Mark

**Comment by prusnak** [comment]
Can you send a pull request so we can see if the build check works?

**Comment by markubiak** [comment]
I can definitely send a pull request, but I think something important to note is that my workaround is exactly that, a workaround, and 100% breaks normal build (no --with-system-hidapi flag). I tried to fix the regression for about a half hour but it's not straightforward. So mainlining that is not a great idea.

**Comment by markubiak** [comment]
I opened #55 primarily to show you what I'm talking about. The setup.py is already a bit messy, and it makes it messier with having 2 include paths for hidapi. But it does allow for a build --with-system-hidapi that does not use any hardcoded include paths. I don't think there's a way to work around this if hidapi is being build side-by-side here. I did check and hidapi with libusb or hidraw is included in all major distributions, so I favor eventually moving away from building hidapi at all. I'm unsure how this will affect other platforms than linux however.

Thoughts appreciated.

**Comment by leinardi** [comment]
This issue makes quite more complex to package this wrapper with flatpak.

**Comment by prusnak** [comment]
I guess it's up to the Flatpak package maintainer to adjust the upstream code to work with Flatpak.

**Comment by jonasmalacofilho** [comment]
Well, the _hardcoded_ path has caused issues for me as well, when packaging code that depends on cython-hidapi for Linuxbrew.

What would be the ideal way to handle headers with Cython?  Maybe check and/or forward `CFLAGS` or something?

**Comment by mcuee** [comment]
Is it possible to make "--with-system-hidapi" the normal build?

**Comment by mcuee** [comment]
From:
* https://github.com/trezor/cython-hidapi/pull/55#issuecomment-1543941318

> This may help.
> https://github.com/matze/pkgconfig/

```
$ python
Python 3.10.10 (main, Feb 10 2023, 19:20:41)  [GCC 12.2.0 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import pkgconfig
>>> pkgconfig.cflags('hidapi')
'-IC:/msys64/mingw64/include/hidapi'
>>> pkgconfig.libs('hidapi')
'-lhidapi'
>>> pkgconfig.parse('hidapi')
defaultdict(<class 'list'>, {'include_dirs': ['C:/msys64/mingw64/include/hidapi'], 'libraries': ['hidapi']})
>>> quit()
```

**Comment by prusnak** [comment]
Should be fixed via #162
